engine/src/main/scala/cromwell/engine/WorkflowState.scala

```diff
@@ -1,42 +1,55 @@
 package cromwell.engine
 
+import scalaz.Semigroup
+
 sealed trait WorkflowState {
   def isTerminal: Boolean
+  def append(that: WorkflowState): WorkflowState
 }
 
 object WorkflowState {
   private lazy val WorkflowState = Seq(WorkflowSubmitted, WorkflowRunning, WorkflowFailed, WorkflowSucceeded, WorkflowAborting, WorkflowAborted)
 
   def fromString(str: String): WorkflowState = WorkflowState.find(_.toString == str).getOrElse(
     throw new NoSuchElementException(s"No such WorkflowState: $str"))
+
+  implicit val WorkflowStateSemigroup = new Semigroup[WorkflowState] {
+    override def append(f1: WorkflowState, f2: => WorkflowState): WorkflowState = f1.append(f2)
+  }
 }
 
 case object WorkflowSubmitted extends WorkflowState {
   override val toString: String = "Submitted"
   override val isTerminal = false
+  override def append(that: WorkflowState): WorkflowState = that
```

Comment:
aside (and it was already like this and I believe you disagree anyways) but cases like this where there are multiple things to override and they're all trivial is exactly where I prefer the fp/pattern match style of ADT-ing vs all of the overrides and the logic in multiple places